### PR TITLE
Write multiple coils functionality to v3 TCP server.

### PR DESCRIPTION
**Pull request by AnthonyMujic**

### CODE REVIEW

**Comment by stefanpoeter** [comment]
I've checked the changes. Everythings fine except that the tests for writing multiple coils fails and the buffer utils module should contain  BufferUtils class with static methods. Fits better to the rest of the code style. I can do it if you want to.

File: src/buffer-utils.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+* Resultant Coils: [0b11110010, 0b11111111, 0b11111111, 0b11111111]
+* The resultant coils are set to shifted outputs, but special attention needs to
+* be paid to the first and last bytes.
+*
+* This code is broken up into 3 funtions:
+*       bufferShift
+*       firstByte
+*       lastByte
+*/
+
```
Comment (stefanpoeter) [inline src/buffer-utils.js]: `class BufferUtils`

File: src/buffer-utils.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+/** bufferShift shift a buffer of ouputs so they can be used to overwrite existing coils
+* @param {start_address} first coil to write
+* @param {end_address} last coil to write
+* @param {outputs} buffer of outputs to write
+* @returns shifted output buffer
+*/
+
+function bufferShift(start_address, end_address, outputs) {
```
Comment (stefanpoeter) [inline src/buffer-utils.js]: `static bufferShift(...)`

File: src/buffer-utils.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  return buffer
+}
+
+/** firstByte ensure first byte is set correctly
+* @param {start_address} first coil to write
+* @param {origianl_byte} byte from the original coils buffer
+* @param {output_byte} first byte from the shifted outputs buffer
+* @returns correct first byte to be written to coils buffer
+*/
+function firstByte(start_address, original_byte, output_byte) {
```
Comment (stefanpoeter) [inline src/buffer-utils.js]: `static firstByte(...)`

File: src/buffer-utils.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    return first_byte
+}
+
+/** lastByte ensure last byte is set correctly
+* @param {end_address} last coil to write
+* @param {origianl_byte} byte from the original coils buffer
+* @param {output_byte} last byte from the shifted outputs buffer
+* @returns correct last byte to be written to coils buffer
+*/
+function lastByte(end_address, original_byte, output_byte) {
```
Comment (stefanpoeter) [inline src/buffer-utils.js]: `static lastByte(...)`

File: src/buffer-utils.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    let end_shift = end_address % 8
+    let mask = 0xff << end_shift
+    let masked_original_byte = original_byte & mask
+
+    let last_byte = output_byte + masked_original_byte
+
+    return last_byte
+}
+
+module.exports = {
```
Comment (stefanpoeter) [inline src/buffer-utils.js]: module.exports BufferUtils

**Comment by AnthonyMujic** [comment]
Sounds good to me. What is the workflow from here? Do I cancel the pull
request, make the commented changes, then do another pull request?

**Comment by stefanpoeter** [comment]
Simply make the changes in the responding repository, commit and push the changes

**Pull request merged**